tests/adc/interface.test.cpp

```diff
@@ -1 +1,52 @@
+#include <boost/ut.hpp>
 #include <libhal/adc/interface.hpp>
+
+namespace hal {
+namespace {
+constexpr auto expected_value = float(0.5);
+
+class test_adc : public hal::adc
+{
+public:
+  float m_returned_position{ 0.5f };
+  bool m_tests_pass{ true };
```

Comment:
```suggestion
  bool m_return_error_status{ true };
```

The previous names gives the impression that this would fail the test and not simply return an error value.

Do the same for the others.